DBZ-8827 Documentation for Debezium Platform

mfvitale opened the pull request:
closes: https://issues.redhat.com/browse/DBZ-8827

The nav section will be then revisited with https://issues.redhat.com/browse/DBZ-8846

roldanbob commented on documentation/modules/ROOT/pages/operations/debezium-platform.adoc:
```diff
@@ -0,0 +1,446 @@
+|offset.database.auth.existingSecret
+|Name of the secret where `username` and `password` - of the database used by the platform for storing the offsets - are stored. If not set `offset.database.auth.username` and `offset.database.auth.password` will be used.
+
+When this is used, you don't need to set `offset.database.auth.username` and `offset.database.auth.password`.
```
```suggestion
If you provide the name of a secret, do not set the `offset.database.auth.username` and `offset.database.auth.password` properties.
```

roldanbob commented on documentation/modules/ROOT/pages/operations/debezium-platform.adoc:
```diff
@@ -0,0 +1,446 @@
+|""
+
+|offset.database.auth.username
+|Username of the database used by the platform for storing the offsets
```
```suggestion
|Username through which the platform connects to the offsets database. 
```

roldanbob commented on documentation/modules/ROOT/pages/operations/debezium-platform.adoc:
```diff
@@ -0,0 +1,446 @@
+|user
+
+|offset.database.auth.password
+|Password of the database used by the platform for storing the offsets
```
```suggestion
|Password for the offsets database user specified by `offset.database.auth.username`.

```

roldanbob commented on documentation/modules/ROOT/pages/operations/debezium-platform.adoc:
```diff
@@ -0,0 +1,446 @@
+|password
+
+|schemaHistory.reusePlatformDatabase
+|Pipelines will use database to store schema history. By default, the database used by the conductor service is used. If you want to use a dedicated one set this property to false
```
```suggestion
|Specifies whether pipelines use the configured platform database to store the schema history. 
To configure pipelines to use a different, dedicated database to store the schema history, set the value to `false`.
```

roldanbob commented on documentation/modules/ROOT/pages/operations/debezium-platform.adoc:
```diff
@@ -0,0 +1,446 @@
+|true
+
+|schemaHistory.database.name
+|Name of the database used by the platform for storing the schema history
```
```suggestion
|Name of the dedicated database where the platform stores the schema history.
```

roldanbob commented on documentation/modules/ROOT/pages/operations/debezium-platform.adoc:
```diff
@@ -0,0 +1,446 @@
+|postgres
+
+|schemaHistory.database.host
+|Host of the database used by the platform for storing the schema history
```
```suggestion
|Host for the dedicated database where the platform stores the schema history.
```

roldanbob commented on documentation/modules/ROOT/pages/operations/debezium-platform.adoc:
```diff
@@ -0,0 +1,446 @@
+|postgres
+
+|schemaHistory.database.port
+|Port of the database used by the platform for storing the schema history
```
```suggestion
|Port through which the platform connects to the dedicated database where it stores the schema history.
```

roldanbob commented on documentation/modules/ROOT/pages/operations/debezium-platform.adoc:
```diff
@@ -0,0 +1,446 @@
+|5432
+
+|schemaHistory.database.auth.existingSecret
+|Name of the secret where `username` and `password` - of the database used by the platform for storing the schema history -are stored. If not set `schemaHistory.database.auth.username` and `schemaHistory.database.auth.password` will be used.
```
```suggestion
|Name of the secret that stores the `username` and `password` that the platform uses to authenticate with the database that stores the schema history. 
If you do not specify value, instead of using a secret to store credentials, the platform uses the values of the schemaHistory.database.auth.username` and `schemaHistory.database.auth.password` properties to authenticate with the database.
```

roldanbob commented on documentation/modules/ROOT/pages/operations/debezium-platform.adoc:
```diff
@@ -0,0 +1,446 @@
+|schemaHistory.database.auth.existingSecret
+|Name of the secret where `username` and `password` - of the database used by the platform for storing the schema history -are stored. If not set `schemaHistory.database.auth.username` and `schemaHistory.database.auth.password` will be used.
+
+When this is used, you don't need to set `schemaHistory.database.auth.username` and `schemaHistory.database.auth.password`.
```
```suggestion
If you provide the name of a secret, do not set the `schemaHistory.database.auth.username` and `schemaHistory.database.auth.password` properties.
```

roldanbob commented on documentation/modules/ROOT/pages/operations/debezium-platform.adoc:
```diff
@@ -0,0 +1,446 @@
+|""
+
+|schemaHistory.database.auth.username
+|Username of the database used by the platform for storing the schema history
```
```suggestion
|Username through which the platform connects to the schema history database.
```

roldanbob commented on documentation/modules/ROOT/pages/operations/debezium-platform.adoc:
```diff
@@ -0,0 +1,446 @@
+|user
+
+|schemaHistory.database.auth.password
+|Password of the database used by the platform for storing the schema history
```
```suggestion
|Password for the schema history database user specified by `schemaHistory.database.auth.username` property.
```

roldanbob commented on documentation/modules/ROOT/pages/operations/debezium-platform.adoc:
```diff
@@ -0,0 +1,446 @@
+|password
+
+|env
+|List of env variable to pass to the conductor
```
```suggestion
|List of environment variables to pass to the conductor.
```

roldanbob reviewed (changes requested):
Reviewed through the Installation instruction (L200). I'll start work on the instructions for using the platform tomorrow.

Naros commented on documentation/modules/ROOT/pages/operations/debezium-platform.adoc:
```diff
@@ -0,0 +1,446 @@
+== Basic concepts
+In {prodname} Management Platform there are four main concepts:
+
+Source:: Defines the source of your data
+Destination:: Defines the destination of your data
+Transform:: Define how a single data is transformed
+Pipeline:: Defines how your data flows from a source to a destination while being transformed along the way.
```
@roldanbob in the DMP (Debezium Management Platform), we decompose the configuration for Debezium Server into reusable parts. 

So what you would normally see defined in `debezium.transforms`, we take each of those and create a standalone component we call `Transform` that defines the transform type and its entire configuration set.  So when the user creates a pipeline, which is the glue of combining a source, sink, and 0 or more transforms, a user can define the transform once and reuse it in multiple pipelines.

mfvitale replied:
@roldanbob what do you think of something like 

`Transform:: Defines how a single data event is transformed as they flow through the pipeline.`

roldanbob replied:
I think that because we're talking about an event, this should read _"Defines how a single data event is transformed as **it** flows through the pipeline."_

Naros commented on documentation/modules/ROOT/pages/operations/debezium-platform.adoc:
```diff
@@ -0,0 +1,446 @@
+The platform is composed of two main components:
+
+Conductor:: The back-end component which provides a set of APIs to orchestrate and control {prodname} deployments.
+Stage:: The front-end component which provides a user interface to interact with the Conductor.
```
```suggestion
Stage:: The front-end component that provides a user interface to interact with the Conductor.
```

Naros commented on documentation/modules/ROOT/pages/operations/debezium-platform.adoc:
```diff
@@ -0,0 +1,446 @@
+|postgres
+
+|database.auth.existingSecret
+|Name of the secret where `username` and `password` - of the database used by the platform - are stored . If empty a secret will be created using the `username` and `password` properties.
```
I believe what @mfvitale is attempting to convey is that if you supply `database.auth.existingSecret`, the username and value are sourced from the secret. However, if this field is left empty and no secret is supplied, the values of `database.auth.username` and `database.auth.password` are used.

roldanbob replied:
Ok, that's what I thought was going on, but the wording here being different than what appears later made me question my understanding. Thanks.

Naros commented on documentation/modules/ROOT/pages/operations/debezium-platform.adoc:
```diff
@@ -0,0 +1,446 @@
+==== Configure the destination with Smart Editor
+The other option is the Smart Editor, where you can directly edit/paste the JSON configuration.
+For those familiar with {prodname}, this is quite similar to {prodname} Server `sink` configuration section with small differences.
+Usually you have that the configuration of a particular sink are prefixed with `debezium.sink.<sink_name>` where `<sink_name` is the sink `type`.
```
```suggestion
Usually you have that the configuration of a particular sink are prefixed with `debezium.sink.<sink_name>` where `<sink_name>` is the sink `type`.
```

Naros commented on documentation/modules/ROOT/pages/operations/debezium-platform.adoc:
```diff
@@ -0,0 +1,446 @@
+Once you are in the `Pipeline designer` you can modify the order of transformations or delete one by clicking on the image:transformation-box-edit.png[alt text] icon.
+
+==== Observability
+Currently, th observability is limited to logs that can be accessed from two places.
```
```suggestion
Observability is currently limited to viewing the Debezium Server logs.
```

Naros reviewed (commented):
I'll leave the remainder to Bob to review with his technical touch, but these are a few things I saw.

mfvitale commented on documentation/modules/ROOT/pages/operations/debezium-platform.adoc:
```diff
@@ -0,0 +1,446 @@
+|""
+
+|stage.image
+|Image for the stage (UI)
```
Effectively this images is used for the `stage` pod and not for build.

mfvitale replied:
Well "stage" is the name of the pod that helm will deploy on K8s using this image.

mfvitale commented:
@roldanbob Thanks for your suggestions. Can you please give another pass?

roldanbob commented:
@mfvitale Just want to let you know that I'm working on this _now_. It's taking me longer than I anticipated, but I should have it finished before I break for lunch.

roldanbob commented on documentation/modules/ROOT/pages/operations/debezium-platform.adoc:
```diff
@@ -0,0 +1,477 @@
+Only the pipeline will be removed, the source, the destination and the transformations will not be deleted.
+
+==== Edit
+To edit a pipeline, go to the `Pipeline` menu and then click the `action` menu of the destination you want to edit, then click `Edit pipeline`.
```
```suggestion
. From the platform UI, open the **Pipeline** menu, click the **action** menu for the pipeline that you want to edit, and then click **Edit pipeline**.
```

roldanbob commented on documentation/modules/ROOT/pages/operations/debezium-platform.adoc:
```diff
@@ -0,0 +1,477 @@
+
+==== Edit
+To edit a pipeline, go to the `Pipeline` menu and then click the `action` menu of the destination you want to edit, then click `Edit pipeline`.
+As first step you can modify the transformations through the pipeline designer, we will go deeper in this part in the next section, and then you can edit the name, the description and the log level.
```
```suggestion
. Use the pipeline designer to modify transformations as needed.
You can edit the name, description, and log level for the transformation.
For more information about using the pipeline designer, see xref:debezium-platform-using-the-pipeline-designer-to-remove-and-order-transformations[Using the Pipeline designer to remove and order transformations].
```

roldanbob commented on documentation/modules/ROOT/pages/operations/debezium-platform.adoc:
```diff
@@ -0,0 +1,477 @@
+As first step you can modify the transformations through the pipeline designer, we will go deeper in this part in the next section, and then you can edit the name, the description and the log level.
+
+===== Remove or ordering transformations
+Once you are in the `Pipeline designer` you can modify the order of transformations or delete one by clicking on the image:transformation-box-edit.png[alt text] icon.
```
```suggestion
From the `Pipeline designer`, you can modify the order of transformations or delete transformations.

* To delete a transformation, click the Transform box (image:transformation-box-edit.png[transformation box]).
```
... and click **Delete**? Or is the deletion done simply by clicking on the box?
And how do you arrange transformation? Is drag-and-drop enabled in the UI?

mfvitale replied:
I'm referring to this 

![image](https://github.com/user-attachments/assets/9af49447-28fa-400b-9950-f736d87a443c)

you click on the "pencil" icon and you will go on the edit

![image](https://github.com/user-attachments/assets/7e9c3eb5-4780-44c4-a5c2-a029729a0457)

roldanbob commented on documentation/modules/ROOT/pages/operations/debezium-platform.adoc:
```diff
@@ -0,0 +1,477 @@
+Once you are in the `Pipeline designer` you can modify the order of transformations or delete one by clicking on the image:transformation-box-edit.png[alt text] icon.
+
+==== Observability
+Observability is currently limited to viewing the Debezium Server logs.
```
```suggestion
The {prodname} platform provides easy access to the {prodname} Server logs.
```

roldanbob commented on documentation/modules/ROOT/pages/operations/debezium-platform.adoc:
```diff
@@ -0,0 +1,477 @@
+
+==== Observability
+Observability is currently limited to viewing the Debezium Server logs.
+You can go to `Pipeline` then click on the pipeline name of your interest and then click on the `Pipeline logs`.
```
```suggestion
. From the platform UI, click **Pipeline**, click the name of the pipeline you want to monitor, and then click **Pipeline logs**.
```

roldanbob commented on documentation/modules/ROOT/pages/operations/debezium-platform.adoc:
```diff
@@ -0,0 +1,477 @@
+==== Observability
+Observability is currently limited to viewing the Debezium Server logs.
+You can go to `Pipeline` then click on the pipeline name of your interest and then click on the `Pipeline logs`.
+The other ways is to go to `Pipeline` and then click the `action` menu of the pipeline you are interested, then click `View logs`.
```
```suggestion

```
I'd suggest removing the alternative method on the principle that it's better for the documentation to be opinionated, and leave it to users to discover shortcuts and alternative methods that are available through mechanisms that we can assume the user is familiar with.

roldanbob reviewed (changes requested):
@mfvitale As per usual, you've provided an impressive and thorough set of information for this new feature. Sorry that I took a while to go through it, but it was a lot to digest, and I'm mostly focused on preparing the docs for the downstream release.
I've indicated some suggestions, mostly for formatting and slight wording changes.  
Long term, we probably want to add anchor IDs throughout. I added just the one on L469, because I thought it was useful to provide a cross-reference to that topic.

The only other general comment I have (I expressed this in one of the specific comments, but I'm repeating it here for visibility) is that to best serve our users, the documentation should provide information that is task vs. UI-based. 

As always, let me know if you have any questions, if you disagree with my suggestions, or if I flat out failed to correctly understand something.

Thanks for the huge effort!

mfvitale commented on documentation/modules/ROOT/pages/operations/debezium-platform.adoc:
```diff
@@ -0,0 +1,446 @@
+|quay.io/debezium/platform-stage:<release_tag>
+
+|conductor.image
+|Image for the conductor
```
Yes, it is.

mfvitale commented on documentation/modules/ROOT/pages/operations/debezium-platform.adoc:
```diff
@@ -0,0 +1,477 @@
+
+|database.auth.existingSecret
+|Name of the secret that stores the `username` and `password` that the platform uses to authenticate with the database. 
+If no value is specified, a secret is created using the `username` and `password` properties.
```
Well the subject that will auto create the secret in this case is helm. So replacing `platform` with `helm` should be fine.

mfvitale commented on documentation/modules/ROOT/pages/operations/debezium-platform.adoc:
```diff
@@ -0,0 +1,477 @@
+When the source is no longer used in any pipeline, you can delete it using the `Delete` option.
+
+==== Edit a source
+To edit a source, go to the `Source` menu and then click the `action` menu of the source you want to edit, then click `Edit`.
```
No, fixed it. Good catch.

mfvitale commented:
Hi @roldanbob thanks for your huge effort in reviewing this. 

I addressed all your suggestions and just two of them remains. Can you give it a look so that this can be ready to be merged for the next release build.

roldanbob reviewed (changes requested):
@mfvitale I think that we're all set now. Get back to me if anything isn't clear.  Thanks!

mfvitale commented:
@roldanbob thanks a lot.
